runtime-library/stdlib/lib_timer.c

```diff
@@ -20,8 +22,13 @@ setTimeout(wasm_exec_env_t exec_env, void *closure, double delay, void *args)
     if (_createTimer != NULL) {
         timer_id = _createTimer(delay);
         bh_hash_map_insert(timer_map, timer_id, closure);
+        wasm_local_obj_ref_t local_ref;
+        wasm_runtime_push_local_object_ref(env, &local_ref);
```

Comment:
This API is not designed for such usage, it can't extend the lifecycle of the object to larger scope.

Reply:
ok, if I want the closure keep alive untill clearTimeout is called, how can I achieve this

Reply:
I think it's better to have a timer library in TS layer, and there will be an array to hold all these closures so that the closure will not be collected. And when the timer is cancelled, remove the closure from the array, then it can be collected.

The native API here don't need to accept the closure any more, the callback dispatch can be finished in the TS timer library.

Reply:
@WenLY1 Here is a simple example to demonstrate the TS layer library.